File: 04__day/backend/models/investigation.py
from typing import Any

from pydantic import BaseModel


class InvestigationPayload(BaseModel):
    pods: dict[str, Any]
    logs: dict[str, Any]
    events: dict[str, Any]
    deployments: dict[str, Any]
    network: dict[str, Any]


class Diagnosis(BaseModel):
    root_cause: str
    explanation: str
    fix: str
    kubectl_command: str
    prevention: str
    confidence: int


class InvestigateResponse(BaseModel):
    status: str
    investigation: InvestigationPayload
    diagnosis: Diagnosis | None = None

File: 04__day/frontend/types/investigation.ts
export interface Diagnosis {
  root_cause: string;
  explanation: string;
  fix: string;
  kubectl_command: string;
  prevention: string;
  confidence: number;
}

export interface ProgressEvent {
  type: "progress";
  step: string;
  status: "in_progress" | "completed";
}

export interface ResultEvent {
  type: "result";
  investigation: unknown;
  diagnosis: Diagnosis | null;
}

export type InvestigationEvent = ProgressEvent | ResultEvent;

export interface InvestigationHistoryItem {
  id: string;
  timestamp: string;
  root_cause: string;
  namespace?: string;
  confidence: number;
  status: string;
  explanation?: string;
  fix?: string;
  kubectl_command?: string;
  prevention?: string;
}

